### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by AbdollahRaafat**
In this project, We add unit tests to our codebase, specifically our classes.

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require '../classes/book'
+
```
Comment (thecodechaser) [inline spec/book_spec.rb]: - [ ] There's a problem with importing classes on test files and I'm getting errors when I run the tests. `require` looks for files from the root directory, not the current directory. To go with the current directory you should use `require_relative` or you can assume importing as from the root directory like `require './classes/book'`. Please fix this issue for all the test files.

image

File: spec/book_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  book = Book.new('The Hobbit', 'J.R.R. Tolkien')
+  book1 = Book.new('Harry Potter', 'J.K. Rowling')
+
+  it 'should create a book' do
+    expect(Book.all.length).to eql(2)
+    expect(Book.all[0].title).to eql('Harry Potter')
+    expect(Book.all[0].author).to eql('J.K. Rowling')
+    expect(Book.all[1].title).to eql('The Hobbit')
+    expect(Book.all[1].author).to eql('J.R.R. Tolkien')
+  end
```
Comment (thecodechaser) [inline spec/book_spec.rb]: - [ ] 5 out of 8 tests are failing, Please make sure that all tests pass successfully.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @XsidX, @AbdollahRaafat, and @Roseokpe👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/classroom_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'spec_helper'
+
+describe Classroom do
```
Comment (FrederickMih) [inline spec/classroom_spec.rb:3]: - [ ] Seems you still have a failing test. Kindly fix :wrench: it.


Screenshot from 2022-10-06 15-48-44

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @AbdollahRaafat , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.




 ## Highlights

-  Gitflow has been followed :+1: 
-  No Linter error :+1: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi Team,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**